consensus/src/main/java/org/apache/iotdb/consensus/ratis/ApplicationStateMachineProxy.java

```diff
@@ -136,11 +141,15 @@ public long takeSnapshot() throws IOException {
     // require the application statemachine to take the latest snapshot
     String metadata = Utils.getMetadataFromTermIndex(lastApplied);
     File snapshotDir = snapshotStorage.getSnapshotDir(metadata);
+    snapshotDir.mkdir();
```

Comment:
if snapshotDir is exist ,  will make dir?

Reply:
Then 'snapshotdir.mkdir ()' will return false and the directory still exists